src/sentry/models/release.py

```diff
@@ -381,61 +381,70 @@ def set_commits(self, commit_list):
                     if not author_email:
                         author = None
                     elif author_email not in authors:
-                        authors[author_email] = author = CommitAuthor.objects.get_or_create(
+                        author_data = {
+                            'name': data.get('author_name')
+                        }
+                        author, created = CommitAuthor.objects.create_or_update(
                             organization_id=self.organization_id,
                             email=author_email,
-                            defaults={
-                                'name': data.get('author_name'),
-                            }
-                        )[0]
-                        if data.get('author_name') and author.name != data['author_name']:
-                            author.update(name=data['author_name'])
+                            defaults=author_data,
+                            values=author_data)
```

Comment:
Minor: You can just pass `author_data` to `values` and have it used in both create/update.

Reply:
We only update `author_name` here if `data['author_name']` is not null. We also skip the update if nothing has changed, not sure this is a huge deal though.

Reply:
We'll emit an additional SQL query, but otherwise it should be a no-op.

Reply:
We're adding the case here where we can now unset the author name, which wouldn't happen before. Not sure it's a problem, just wanted to point it out.